src/test/java/org/jabref/model/entry/identifier/DOITest.java

```diff
@@ -306,4 +306,15 @@ public void rejectMissingDivider() {
     public void rejectMissingDividerInShortDoi() {
         assertThrows(IllegalArgumentException.class, () -> new DOI("10gf4gqc end"));
     }
+
+    @Test
+    public void acceptEncodedDoiUrl() {
+        assertEquals("10.1175/1520-0493(2002)130<1913:EDAWPO>2.0.CO;2", new DOI("https://doi.org/10.1175/1520-0493(2002)130<1913:EDAWPO>2.0.CO;2").getDOI());
+        assertEquals("10.1175/1520-0493(2002)130<1913:EDAWPO>2.0.CO;2", new DOI("https://doi.org/10.1175/1520-0493(2002)130%3C1913:EDAWPO%3E2.0.CO;2").getDOI());
```

Comment:
This should go to a parameterized test, `@CsvSource` (https://mikemybytes.com/2021/10/19/parameterize-like-a-pro-with-junit-5-csvsource/)

Reply:
Oh, we do have `testData()` very up in the class. Please use that!

Reply:
Thank you very much for your feedback!

I have now pushed my changes concerning:

1. /DOI.java:130: Prefer `java.net.URLDecoder.decode(String, Charset)`
2. Moving suggested tests in `testData()`

In addition I have executed `rewriteRun` task from gradle